#### Customer Feedback Sentiment Analysis for ShopEase Europe
By
Olayemi Balogun

#### Business Introduction
ShopEase Europe is a large online shopping company that sells different products across Europe.
Every day, customers leave reviews, comments, and complaints on the website, social media, and customer support channels. With over a million users each month, there is too much feedback for staff to read manually.
The company wants to understand what customers are saying and how they feel. To maintain high customer satisfaction and improve service delivery.

#### Problem Statement
- Thousands of reviews are received every day, making it difficult for employees to read and analyze them all.
- Different analysts may interpret the same review differently, leading to inconsistent results.
- Although feedback is available, it takes too much time to turn it into useful business actions.

#### Project Objectives

- Clean and prepare customer reviews written in different languages.
- Build machine learning models that can predict customer sentiment.
- Identify the main reasons customers are happy or unhappy.
- Create summaries of customer opinions.
- Display results in an interactive dashboard.

##### Import the libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from langdetect import detect
from langdetect.lang_detect_exception import LangDetectException
import re
import spacy
import nltk
from nltk.corpus import stopwords
import plotly.express as px

nltk.download('punkt')
nltk.download('stopwords')


[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\balog\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\balog\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


True

##### Load the dataset

In [2]:
df = pd.read_csv(r'C:\Users\balog\OneDrive\Documents\Olayemi\Intership\Customer Feedback Analysis\Data\reviews.csv')
df.head()

,review_id,product_category,timestamp,country,rating,review,sentiment
0,REV-50BCBCD9,Sports,2024-09-16 13:44:26+00:00,US,1,"I registered on the website, tried to order a ...",Negative
1,REV-6D2B2651,Toys,2024-09-16 18:26:46+00:00,GB,1,Had multiple orders one turned up and driver h...,Negative
2,REV-F7E80372,Toys,2024-09-16 21:47:39+00:00,GB,1,I informed these reprobates that I WOULD NOT B...,Negative
3,REV-ED2B173F,Sports,2024-09-17 07:15:49+00:00,AU,1,I have bought from Amazon before and no proble...,Negative
4,REV-E48A7AB9,Fashion,2024-09-16 18:37:17+00:00,GB,1,If I could give a lower rate I would! I cancel...,Negative


##### Data Overview

In [3]:
print("="*50)
print("DATASET OVERVIEW")
print("="*50)

print(f"Rows: {df.shape[0]}")
print(f"Columns: {df.shape[1]}")
print(f"Shape: {df.shape}")

print("\nColumn Names:")
print(df.columns.tolist())

print("\nData Types:")
print(df.dtypes)

print("\nMissing Values:")
print(df.isnull().sum())

print("\nDuplicate Rows:")
print(df.duplicated().sum())

print("\nUnique Values:")
print(df.nunique())

for col in df.columns:
    print(f"\n{col}")
    print(df[col].unique())

print("\nFirst 5 Rows:")
print(df.head())


DATASET OVERVIEW
Rows: 21055
Columns: 7
Shape: (21055, 7)

Column Names:
['review_id', 'product_category', 'timestamp', 'country', 'rating', 'review', 'sentiment']

Data Types:
review_id           object
product_category    object
timestamp           object
country             object
rating               int64
review              object
sentiment           object
dtype: object

Missing Values:
review_id           0
product_category    0
timestamp           0
country             1
rating              0
review              0
sentiment           0
dtype: int64

Duplicate Rows:
0

Unique Values:
review_id           21055
product_category        7
timestamp           21054
country               148
rating                  5
review              20407
sentiment               3
dtype: int64

review_id
['REV-50BCBCD9' 'REV-6D2B2651' 'REV-F7E80372' ... 'REV-9305C28A'
 'REV-40867447' 'REV-4765654E']

product_category
['Sports' 'Toys' 'Fashion' 'Beauty' 'Food & Grocery' 'Home & Living'
 'Electroni

       OVERVIEW
     - The dataset contains 21,055 rows (customer reviews) and 7 columns.
     - 1 missig value in the column 'country' and no duplicated rows.
     - There are 148 countries, 7 product categories, rating is from 1 to 5, 20,407 unique review texts.
     - The timestamp column is stored as an object instead of a date format.
     - Clean sentiment labels and no inconsistency.
     - Multilingual reviews.

##### Data Report

In [4]:
data_report = pd.DataFrame({
    "Column Name": df.columns,
    "Data Type": df.dtypes.values,
    "Missing Values": df.isnull().sum().values,
    "Unique Values": df.nunique().values,
    "Data Quality Issues": ["No issues", "No inconsistencies", "Convert to datetime", "Standardized codes", 
                     "Valid values", "Repetitive texts", "Validated labels"] 
})

data_report

,Column Name,Data Type,Missing Values,Unique Values,Data Quality Issues
0,review_id,object,0,21055,No issues
1,product_category,object,0,7,No inconsistencies
2,timestamp,object,0,21054,Convert to datetime
3,country,object,1,148,Standardized codes
4,rating,int64,0,5,Valid values
5,review,object,0,20407,Repetitive texts
6,sentiment,object,0,3,Validated labels
